Automattic/wp-calypso
# Ensure that all email management views send Tracks page views

#### Proposed Changes

* Send Tracks page view events for all email management pages (two pages were previously missing these).
* Lift `PageViewTracker` components higher up in the component tree. There is a fairly intricate co-dependence between `index.js`, `controller.js`, `paths.js` and the different page components. To reduce the components' dependence on `paths.js`, I found it logical to put `PageViewTracker` at the `controller.js` level. This is already being practiced in [`calypso/my-sites/domains/controller.jsx`](https://github.com/Automattic/wp-calypso/blob/trunk/client/my-sites/domains/controller.jsx)


#### Testing Instructions



Most email management views were already submitting page view events, but since I've modified the placement of `PageViewTracker`, I have tested all affected views within the email management section.

The [Tracks Vigilante browser extension](https://github.com/Automattic/tracks-chrome-extension) is helpful when testing these changes – it sniffs the network traffic and displays Tracks requests in a more readable manner than the DevTools Network panel.

To test these changes, ensure that each of the following slugs submit a `calypso_page_view` event:

##### Doesn't require an existing subscription
/email/:domain/purchase/:site
/email/:domain/compare/:site

##### Requires any subscription
/email/:site
/email/:domain/manage/:site

##### Requires a Google Workspace domain
/email/:domain/google-workspace/add-users/:site

##### Requires a Titan domain
/email/:domain/titan/manage-mailboxes/:site
/email/:domain/titan/new/:site
/email/:domain/titan/set-up-mailbox/:site
/email/:domain/titan/manage/:site (**did not submit page view events previously**)
/email/:domain/titan/set-up-mailbox/thank-you/:site (**did not submit page view events previously**)

##### Requires an email forwarding domain
/email/:domain/forwarding/add/:site
